modules/rank.js

```diff
@@ -0,0 +1,164 @@
+
+      if (subcommand === "view") {
+        // View member rankings
+        await interaction.deferReply();
+        const members = interaction.guild.members.cache;
+        const member = interaction.options.getMember("user") ?? interaction.member;
+        const leaderboard = await Module.db.user.getLeaderboard({ members, member });
+        const record = leaderboard.find(r => r.discordId == member.id);
+
+        if (record) {
+          const level = Rank.level(record.totalXP);
+          const nextLevel = Rank.minXp(level + 1).toLocaleString();
+
+          const embed = u.embed({ author: member })
+          .setTitle("LDSG Season Chat Ranking")
+          .setURL("https://my.ldsgamers.com/leaderboard")
+          .setFooter("https://my.ldsgamers.com/leaderboard")
+          .addField("Rank", `Season: ${record.rank} / ${members.size}`, true)
+          .addField("Level", `Current Level: ${level.toLocaleString()}\nNext Level: ${nextLevel} XP`, true)
+          .addField("Exp.", `Season: ${record.currentXP.toLocaleString()} XP\nLifetime: ${record.totalXP.toLocaleString()} XP`, true);
+
+          await interaction.editReply({ embeds: [embed] });
+        } else {
+          const snark = [
+            "don't got time for dat.",
+            "ain't interested in no XP gettin'.",
+            "don't talk to me no more, so I ignore 'em."
+          ];
+          await interaction.editReply(`**${member}** ${u.rand(snark)}\n(Try \`/rank track\` if you want to participate in chat ranks!)`);
+        }
+      } else if (subcommand === "leaderboard") {
+        // View leaderboard
+        await interaction.deferReply();
+
+        const members = interaction.guild.members.cache;
+
+        const leaderboard = await Module.db.user.getLeaderboard({
+          members,
+          member: interaction.member
+        });
+
+        const records = leaderboard.map(l => `${l.rank}: ${members.get(l.discordId).toString()} (${l.currentXP.toLocaleString()} XP)`);
+        const embed = u.embed()
+        .setTitle("LDSG Season Chat Leaderboard")
+        .setThumbnail(interaction.guild.iconURL({ format: "png" }))
+        .setURL("https://my.ldsgamers.com/leaderboard")
+        .setDescription("Current season chat rankings:\n" + records.join("\n"))
+        .setFooter("Use `/rank track` to join the leaderboard!");
+
+        await interaction.editReply({ embeds: [embed] });
+      } else if (subcommand === "track") {
+        // Set XP tracking
+        await interaction.deferReply({ ephemeral: true });
+        const track = interaction.options.getBoolean("choice");
+        await Module.db.user.trackXP(interaction.user, track);
+        await interaction.editReply({
+          content: `Ok! I'll ${track ? "start" : "stop"} tracking your XP!`,
+          ephemeral: true
+        });
+      } else {
+        interaction.reply({
+          content: "Well, this is embarrasing. I don't know what you asked for.",
+          ephemeral: true
+        });
```

Comment:
This *should* never happen; probably worth quietly throwing an error here as well.

Reply:
Agreed it shouldn't ever happen, it was a failsafe. Agreed that an error is a good call.

Reply:
I'm all for failsafes! 👍

(side note: Chrome spell checker has no problem with failsafe, but it doesn't like failsafes)